database-migration/src/test/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractStatisticsFromVariantTest.java

```diff
@@ -0,0 +1,209 @@
+        MongoTemplate mongoTemplate = new MongoTemplate(new Fongo("testServer").getMongo(), "variantWithStatistics");
+        DBCollection variantsCollection = mongoTemplate.getCollection(VARIANT_COLLECTION_NAME);
+        DBCollection statisticsCollection = mongoTemplate.getCollection(STATISTICS_COLLECTION_NAME);
+
+        DBObject variantWithSt = (DBObject) JSON
+                .parse("{ '_id' : 'Chr10_10000022_A_G', 'chr' : 'Chr10', 'start' : 10000022, 'files' : [ { 'fid' : 'ERZ123186', 'sid' : 'PRJEB10964', 'attrs' : { 'QUAL' : '255.0', 'CNV' : '64', 'TA' : 'Intergenic'}, 'fm' : 'GT:GL:GP:GQ:DP:AAC:LP', 'samp' : { '0/0' : [ 3, 21, 28, 35, 41, 42, 43, 46, 48, 51, 67, 68,  89, 98, 103 ], 'def' : '-1/-1', '0/1' : [ 36, 97 ], '1/1' : [ 13, 14, 16, 17, 18, 19, 33, 34, 37, 38, 39, 40, 44, 45, 47, 50, 52, 53, 55, 56, 58, 61, 62, 64, 70, 76, 83, 84, 85, 87, 88, 90, 92, 93, 96, 99, 100, 101 ] } } ], 'ids' : [ ], 'type' : 'SNV', 'end' : 10000022, 'len' : 1, 'ref' : 'A', 'alt' : 'G', '_at' : { 'chunkIds' : [ 'Chr10_10000_1k', 'Chr10_1000_10k' ] }, 'hgvs' : [ { 'type' : 'genomic', 'name' : 'Chr10:g.10000022A>G' } ], 'st' : [ { 'maf' : 0.3709677457809448, 'mgf' : 0.032258063554763794, 'mafAl' : 'A', 'mgfGt' : '0/1', 'missAl' : 84, 'missGt' : 42, 'numGt' : { '1/1' : 38, '0/1' : 2, '0/0' : 22, '-1/-1' : 42 }, 'cid' : 'ALL', 'sid' : 'PRJEB10964', 'fid' : 'ERZ123186' } ] }");
+        variantsCollection.insert(variantWithSt);
+
+        extractStatisticsFromVariant.migrateStatistics(mongoTemplate);
+        extractStatisticsFromVariant.removeStatisticsFromVariants(mongoTemplate);
+
+        assertEquals(1, variantsCollection.count());
+        DBObject variantObj = variantsCollection.findOne();
+        assertNull(variantObj.get("st"));
+        assertNotNull(variantObj.get("maf"));
+        double newMafInVariant = (Double) ((BasicDBList) variantObj.get("maf")).get(0);
+        double mafInStVariant = (Double) ((BasicDBObject) ((BasicDBList) variantWithSt.get("st")).get(0)).get("maf");
+
+        assertEquals(mafInStVariant, newMafInVariant, 0);
+
+        assertEquals(1, statisticsCollection.count());
+        DBObject statisticsObj = statisticsCollection.findOne();
+        assertNull(statisticsObj.get("fid"));
+
+        assertNotNull(statisticsObj.get("vid"));
+        assertNotNull(statisticsObj.get("chr"));
+        assertNotNull(statisticsObj.get("start"));
+        assertNotNull(statisticsObj.get("ref"));
+        assertNotNull(statisticsObj.get("alt"));
+        assertNotNull(statisticsObj.get("cid"));
+        assertNotNull(statisticsObj.get("sid"));
+        assertNotNull(statisticsObj.get("maf"));
+        assertNotNull(statisticsObj.get("mgf"));
+        assertNotNull(statisticsObj.get("mafAl"));
+        assertNotNull(statisticsObj.get("mgfGt"));
+        assertNotNull(statisticsObj.get("missAl"));
+        assertNotNull(statisticsObj.get("missGt"));
+        assertNotNull(statisticsObj.get("numGt"));
```

Comment:
It is possible to compare 2 DBObjects for equality, which would ensure the structure **and** the value types are fine.